ADD documentation/Tutorial for learn then test (#347)
scikit-learn-contrib/MAPIE: doc/theoretical_description_multilabel_classification.rst

- LTT:

.. math::

>     \mathbb{P}(R(\mathcal{T}_\lambda) \leq \alpha ) \geq 1 - \delta & p_\lambda \leq \frac{\delta}{\lvert \Lambda \rvert}
> 

Reviewer: ```suggestion
    \mathbb{P}(R(\mathcal{T}_{\hat{\lambda}}) \leq \alpha ) \geq 1 - \delta \quad \texttt{with} \quad p_\lambda \leq \frac{\delta}{\lvert \Lambda \rvert}
```

Author: Done

3. Learn Then Test
------------------

3.1. General settings
---------------------
The settings here are the same as RCPS and CRC, we just need to introduce some new parameters:

- Let :math:`\Lambda` be a discretized for our :math:`\lambda`, meaning that :math:`\Lambda = \{\lambda_1, ..., \lambda_n\}`.

- Let :math:`p_\lambda` be a valid p-value for the null hypothesis :math:`\mathbb{H}_j: R(\lambda_j)>\alpha`.

The goal of this method is to control any loss whether monotonic, bounded or not, by performing risk control through multiple
hypothesis testing. We can express the goal of the procedure as follows:

.. math::
    \mathbb{P}(R(\mathcal{T}_{\lambda}) \leq \alpha ) \geq 1 - \delta

In order to find all the parameters :math:`\lambda` that satisfy the above condition, the Learn Then Test framework proposes to do the following:

> - First across the collections of functions :math:`(T_\lambda)_{\lambda\in\Lambda}`, we estimate the risk on the calibration data
> :math:`\{(x_1, y_1), \dots, (x_n, y_n)\}`.

Reviewer: ```suggestion
- First across the collections of functions :math:`(T_\lambda)_{\lambda\in\Lambda}`, we estimate the risk on the calibration data
  :math:`\{(x_1, y_1), \dots, (x_n, y_n)\}`.
```

Author: Done

> - For each :math:`\lambda_j` in a discrete set :math:`\Lambda = \{\lambda_1, \lambda_2,\dots, \lambda_n\}`, we associate the null hypothesis
> :math:`\mathbb{H}_j: R(\lambda_j) > \alpha`, as rejecting the hypothesis corresponds to selecting :math:`\lambda_j` as a point where risk the risk 

Reviewer: ```suggestion
- For each :math:`\lambda_j` in a discrete set :math:`\Lambda = \{\lambda_1, \lambda_2,\dots, \lambda_n\}`, we associate the null hypothesis
  :math:`\mathbb{H}_j: R(\lambda_j) > \alpha`, as rejecting the hypothesis corresponds to selecting :math:`\lambda_j` as a point where risk the risk 
```

Author: Done add space

is controlled.

> - For each null hypothesis, we compute a valid p-value using a concentration inequality :math:`p_{\lambda_j}`. Here we choose to compute the Hoeffding-Bentkus p-value
> introduced in the paper [3].

Reviewer: ```suggestion
- For each null hypothesis, we compute a valid p-value using a concentration inequality :math:`p_{\lambda_j}`. Here we choose to compute the Hoeffding-Bentkus p-value
  introduced in the paper [3].
```

Author: done add space

> - Return :math:`\hat{\Lambda} =  \mathcal{A}(\{p_j\}_{j\in\{1,\dots,\lvert \Lambda \rvert})`, where :math:`\mathcal{A}`, is an algorithm
> that controls the family-wise-error-rate (FWER), for example bonferonni correction.

Reviewer: ```suggestion
- Return :math:`\hat{\Lambda} =  \mathcal{A}(\{p_j\}_{j\in\{1,\dots,\lvert \Lambda \rvert})`, where :math:`\mathcal{A}`, is an algorithm
  that controls the family-wise-error-rate (FWER), for example bonferonni correction.
```

Author: Done add space

3. Learn Then Test
------------------

3.1. General settings
---------------------
We are going to present the Learn Then Test framework
The settings here are the same as RCPS and CRC, we just need to introduce some new parameters:

- Let :math:`\Lambda` be a discretized for our :math:`\lambda`, meaning that :math:`\Lambda = \{\lambda_1, ..., \lambda_n\}`.

- Let :math:`p_\lambda` be a valid p-value for the null hypothesis :math:`\mathbb{H}_j: R(\lambda_j)>\alpha`.

The goal of this method is to control any loss whether monotonic, bounded or not, by performing risk control through multiple
hypothesis testing. We can express the goal of the procedure as follows:

.. math::
    \mathbb{P}(R(\mathcal{T}_{\lambda}) \leq \alpha ) \geq 1 - \delta

In order to find all the parameters :math:`\lambda` that satisfy the above condition, the Learn Then Test framework proposes to do the following:

- First across the collections of functions :math:`(T_\lambda)_{\lambda\in\Lambda}`, we estimate the risk on the calibration data
  :math:`\{(x_1, y_1), \dots, (x_n, y_n)\}`.

> - For each :math:`\lambda_j` in a discrete set :math:`\Lambda = \{\lambda_1, \lambda_2,\dots, \lambda_n\}`, we associate the null hypothesis
>   :math:`\mathcal{H}_j: R(\lambda_j) > \alpha`, as rejecting the hypothesis corresponds to selecting :math:`\lambda_j` as a point where risk the risk 
> is controlled.

Reviewer: ```suggestion
- For each :math:`\lambda_j` in a discrete set :math:`\Lambda = \{\lambda_1, \lambda_2,\dots, \lambda_n\}`, we associate the null hypothesis
  :math:`\mathcal{H}_j: R(\lambda_j) > \alpha`, as rejecting the hypothesis corresponds to selecting :math:`\lambda_j` as a point where risk the risk 
  is controlled.
```

3. Learn Then Test
------------------

> 3.1. General settings

Author: References are at the end of the document!

> 3.1. General settings

Reviewer: The suggestion was to introduce this part with the reference of the LLT method :)

> 3.1. General settings

Author: Well isn't it better to have all references at the end ?

> In order to find all the parameters :math:`\lambda` that satisfy the above condition, Learn Then Test propose to do the following:

Reviewer 2: In order to find all the parameters :math:`\lambda` that satisfy the above condition, the Learn Then Test framework proposes to do the following:

Reviewer: ```suggestion
In order to find all the parameters :math:`\lambda` that satisfy the above condition, the Learn Then Test framework proposes to do the following:
```

Author: done

> 1: First across the collections of functions :math:`(T_\lambda)_{\lambda\in\Lambda}`, we estimate the risk on the calibration data

Reviewer 2: These are not bullet points, not sure if that's the aim, but I don't find it very easy to read. Maybe put the numbers in bold?

Author: Done, adding bullet points

Reviewer: Be careful some trouble in the documentation (indentation).

:math:`\{(x_1, y_1), \dots, (x_n, y_n)\}`.

2: For each :math:`\lambda_j` in a discrete set :math:`\Lambda = \{\lambda_1, \lambda_2,\dots, \lambda_n\}`, we associate the null hypothesis
:math:`\mathbb{H}_j: R(\lambda_j) > \alpha`, as rejecting the hypothesis corresponds to selecting :math:`\lambda_j` as a point where risk the risk 
is controlled.

> 3: For each null hypothesis, we compute a valid p-value using a concentration inequality :math:`p_\lambda_j`. Here we choose to compute the Hoeffding-Bentkus p-value

Reviewer 2: There is an error in the run of this code (check build of the documentation).

Reviewer: ```suggestion
3: For each null hypothesis, we compute a valid p-value using a concentration inequality :math:`p_{\lambda_j}`. Here we choose to compute the Hoeffding-Bentkus p-value
```

Author: Done

> 3: For each null hypothesis, we compute a valid p-value using a concentration inequality :math: `p_\lambda_j`. Here we choose to compute the Hoeffding-Bentkus p-value

Reviewer: ```suggestion
3: For each null hypothesis, we compute a valid p-value using a concentration inequality :math:`p_\lambda_j`. Here we choose to compute the Hoeffding-Bentkus p-value
```

Author: Done
